spec/requests/admin/ccms_queues_controller_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe Admin::CCMSQueuesController do
+  let(:admin_user) { create(:admin_user) }
+
+  describe "GET index" do
+    subject(:get_index) { get admin_ccms_queues_path }
+
+    before do
+      sign_in admin_user
+    end
+
+    it "renders successfully" do
+      get_index
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "displays page title" do
+      get_index
+      expect(response.body).to include("Incomplete CCMS Submissions")
+    end
+
+    context "when there are no applications on the queue" do
+      it "has a link to the download csv path" do
+        get_index
+        expect(response.body).to include("Queue is empty")
+      end
+    end
+
+    context "when an application is on the queue" do
+      let!(:ccms_submission) { create(:ccms_submission, :case_ref_obtained) }
+
+      it "has a link to the application" do
+        get_index
+        expect(response.body).to include(admin_ccms_queue_path(ccms_submission.id))
+      end
+    end
+  end
+
+  describe "GET show" do
+    subject(:get_show) { get admin_ccms_queue_path(ccms_submission.id) }
+
+    let!(:ccms_submission) { create(:ccms_submission, :case_ref_obtained) }
+
+    before do
+      sign_in admin_user
+      get_show
+    end
+
+    it "renders successfully" do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "displays ccms case reference" do
+      expect(response.body).to include(ccms_submission.case_ccms_reference)
+    end
+  end
+
+  describe "GET reset_and_restart" do
+    subject(:get_reset_and_restart) { get reset_and_restart_admin_ccms_queue_path(ccms_submission.id) }
+
+    before { sign_in admin_user }
+
+    let!(:ccms_submission) { create(:ccms_submission, :case_ref_obtained) }
+
+    it "calls submission.complete_restart!" do
+      expect(get_reset_and_restart).to redirect_to(admin_ccms_queue_path(ccms_submission.id))
+    end
+  end
+
+  describe "GET restart_current_submission" do
+    subject(:get_restart_current_submission) { get restart_current_submission_admin_ccms_queue_path(ccms_submission.id) }
+
+    let!(:ccms_submission) { create(:ccms_submission, :case_ref_obtained) }
+
+    before { sign_in admin_user }
+
+    it "calls submission.restart_current_submission" do
+      expect(get_restart_current_submission).to redirect_to(admin_ccms_queue_path(ccms_submission.id))
+      expect(flash[:notice]).to match(ccms_submission.id)
+    end
+  end
```

Comment:
This is something for wider discussion (outside the scope of this PR) but it's something I've been thinking about for a while and this is a nice example.

It feels like sometimes our tests are a means to an end (e.g 100% code coverage), but I think we could perhaps be writing more valuable specs to test the behaviour of our application.

In this instance, I think a system spec would be more helpful.

In my opinion, it would naturally
- exercise more functionality (e.g we don't need to explicitly test the index page)
- reflect more accurately the use case
- help us write more accessible code (since we are being more specific and intentional with how we interact with the page, and considering how functionality affects the user directly)

```ruby
require "rails_helper"

RSpec.describe "Admin users can manage the CCMS submission queue", type: :system do
  context "when restarting a submission" do
    it "restarts the submission at the current step" do
      admin_user = create(:admin_user)
      ccms_submission = create(:ccms_submission)
      sign_in admin_user

      visit admin_ccms_queues_path

      within "table" do  # I'd use a more specific helper 
        click_link ccms_submission.case_reference
      end
      
      click_button "Restart at current step"
      
      expect(current_path).to eq(admin_ccms_queue_path(ccms_submission.id))
      expect(page).to have_flash_notice("What the message was")
    end
  end
end
```

I'd be interested to hear what the rest of the team think.

Reply:
Added a link to this conversation to a future dev-time discussion

Reply:
outside of scope for this PR as you mention but would be happy to try system tests out, in place of cucumber.